dbos/workflow.go

```diff
@@ -1228,6 +1244,11 @@ func RetrieveWorkflow[R any](ctx DBOSContext, workflowID string) (workflowPollin
 	if ctx == nil {
 		return workflowPollingHandle[R]{}, errors.New("dbosCtx cannot be nil")
 	}
+
+	// Register the output for gob encoding
+	var r R
+	gob.Register(r)
```

Comment:
We need to perform gob encoding here, because the client might not be running in the same program where the workflow was registered -- and automatic registration was performed -- and I eventually want to move the encoder into the `DBOSContext`.